### PR TITLE
Reduce malloc() and free() call when call dnode_peer_for_key_on_rack()

### PR DESCRIPTION
**Pull request by lampmanyao**

### CODE REVIEW

File: src/dyn_token.h [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 struct dyn_token {
     uint32_t signum;
-    uint32_t *mag;
+    uint32_t mag[4];
```
Comment (shailesh33) [inline src/dyn_token.h]: I have not looked at this code for a while. What do we store in mag? is 4 enough?

File: src/dyn_token.c [hunk 2]
```diff
@@ -35,16 +35,12 @@ void
 init_dyn_token(struct dyn_token *token)
 {
 	token->signum = 0;
-	token->mag = NULL;
```
Comment (shailesh33) [inline src/dyn_token.c]: Initialize mag here?

**Comment by shailesh33** [comment]
Overall it looks good. thanks a lot.

File: src/dyn_token.h (hunk 1 above)
Comment (lampmanyao) [inline src/dyn_token.h]: We store dnode's token or hash value of key in mag.
All of hash functions use 4 bytes, except hash_murmur3(), which uses 128-bit storage.
I think 4 is enough.

Comment (shailesh33) [inline src/dyn_token.h]: Appreciate. Since you are at it, can you ASSERT when it goes beyond 4 bytes?

**Pull request merged**